### ISSUE TITLE
False positive on PEP695 generic class syntax

### ISSUE THREAD

**Issue by Microwave-WYB** [pylint-dev/pylint#9682]
### Bug description

```python
class Updatable[T](ABC):
    """A record that can be updated with a new value."""

    @abstractmethod
    def update(self, value: T) -> Self:
        """Update the record with a new value."""
```

Pylint reports the typevar `T` in `value: T` as an undefined variables

This should be a valid generic typing syntax since 3.12

### Configuration

```ini
[MESSAGES CONTROL]
disable=too-many-arguments,too-few-public-methods,missing-module-docstring,missing-class-docstring,missing-function-docstring
```

### Command used

```shell
pylint --rcfile=.pylintrc qcat
```

### Pylint output

```python
************* Module qcat.trading.core
qcat/trading/core.py:38:28: E0602: Undefined variable 'T' (undefined-variable)

------------------------------------------------------------------
Your code has been rated at 9.86/10 (previous run: 8.99/10, +0.87)
```

### Expected behavior

`T` should be a valid typevar

### Pylint version

```shell
pylint 3.2.2
astroid 3.2.2
Python 3.12.3 (main, Apr 23 2024, 09:16:07) [GCC 13.2.1 20240417]
```
```


### OS / Environment

ArchLinux

### Additional dependencies

_No response_

**Comment by jacobtylerwalls** [comment]
Duplicate of #9335
